site/src/modules/resources/DownloadAgentLogsButton.tsx

```diff
@@ -0,0 +1,64 @@
+import DownloadOutlined from "@mui/icons-material/DownloadOutlined";
+import Button from "@mui/material/Button";
+import { saveAs } from "file-saver";
+import { useState, type FC } from "react";
+import { useQueryClient } from "react-query";
+import { agentLogs } from "api/queries/workspaces";
+import type { WorkspaceAgent, WorkspaceAgentLog } from "api/typesGenerated";
+import { displayError } from "components/GlobalSnackbar/utils";
+
+type DownloadAgentLogsButtonProps = {
+  workspaceId: string;
+  agent: Pick<WorkspaceAgent, "id" | "name" | "status" | "lifecycle_state">;
+  download?: (file: Blob, filename: string) => void;
+};
+
+export const DownloadAgentLogsButton: FC<DownloadAgentLogsButtonProps> = ({
+  workspaceId,
+  agent,
+  download = saveAs,
+}) => {
+  const queryClient = useQueryClient();
+  const isConnected = agent.status === "connected";
+  const [isDownloading, setIsDownloading] = useState(false);
+
+  const fetchLogs = async () => {
+    const queryOpts = agentLogs(workspaceId, agent.id);
+    let logs = queryClient.getQueryData<WorkspaceAgentLog[]>(
+      queryOpts.queryKey,
+    );
+    if (!logs) {
+      logs = await queryClient.fetchQuery(queryOpts);
+    }
+    return logs;
+  };
+
+  return (
+    <Button
+      startIcon={<DownloadOutlined />}
+      disabled={!isConnected || isDownloading}
+      variant="text"
+      size="small"
+      onClick={async () => {
+        try {
+          setIsDownloading(true);
+          const logs = await fetchLogs();
+          if (!logs) {
+            displayError("Failed to fetch logs");
+            setIsDownloading(false);
+            return;
```

Comment:
Small suggestion, could just throw and let the `catch` take care of displaying the error.

```suggestion
            throw new Error("No logs in response")
```